Use regex and find-and-replace to change parameter names in bulk

sciwhiz12 opened the pull request:
Using the power of regular expressions and find-and-replace, we can change parameter names in bulk for names which are repeated very often. This saves a good chunk of time in the planned parameter names audit.

Each find-and-replace operation will be done as one commit, to partially ease review of the changes.

## Current changes:

 - Replace all occurences of `pWorld` with `pLevel`
   - _summary:_ [`regex-01-world-level.md`](https://gist.github.com/sciwhiz12/2c2adab7942f5c3da43ea268dc22ce85#file-regex-01-world-level-md)
 - Replace all occurences of `World` with `Level`
   - _summary:_ [`regex-02-world-level-rest.md`](https://gist.github.com/sciwhiz12/2c2adab7942f5c3da43ea268dc22ce85#file-regex-02-world-level-rest-md)
 - Replace occurrences of `TileEntity` with `BlockEntity`
   - _summary:_ [`regex-03-tilentity-blockentity.md`](https://gist.github.com/sciwhiz12/2c2adab7942f5c3da43ea268dc22ce85#file-regex-03-tilentity-blockentity-md)
 - Replace occurences of `Te` with `BlockEntity`
   - _summary:_ [`regex-04-te-blockentity.md`](https://gist.github.com/sciwhiz12/2c2adab7942f5c3da43ea268dc22ce85#file-regex-04-te-blockentity-md)
 - Remove `In` suffix
   - _summary:_ [`regex-05-in.md`](https://gist.github.com/sciwhiz12/2c2adab7942f5c3da43ea268dc22ce85#file-regex-05-in-md)
   - has thirteen (13) validation errors
 - Add `Value` suffix to otherwise-invalid parameter names due to keywords
   - _summary:_ [`regex-06-value-suffix.md`](https://gist.github.com/sciwhiz12/6966cb397f2576643b50d9ef5d574c3a#file-regex-06-value-suffix-md)
   - fixes twelve (12) validation errors, one (1) validation error remaining
 - Fix parameter name to `pInput`
   - manually applied
   - fixes one (1) validation error, no validation errors remain
 - Replace occurrences of `Placement` to `Decorator`
   - _summary:_ [`regex-07-placement-decorator.md`](https://gist.github.com/sciwhiz12/6966cb397f2576643b50d9ef5d574c3a#file-regex-07-placement-decorator)

sciwhiz12 committed:
Use regex find-and-replace to change `pWorld` to `pLevel`

mcenderdragon commented on data/classes/aos.json:
```diff
@@ -39,7 +39,7 @@
         {
           "index": 2,
-          "name": "pWorldTime"
+          "name": "pLevelTime"
```
is this change valid?

sciwhiz12 replied:
If we're talking about if it meets the standards, then yes it is a valid parameter name.

If we're talking about if `pWorldTime` would more accurately describe the parameter instead of `pLevelTime`, then I have no answer. The point of the PR is to do bulk renaming of parameter names in preparation of a planned parameter names and javadocs audit/review, such as renaming parameters that reference MCP class names (`World`) to their Official/Mojmap class name counterpart (`Level`). _(BTW, there is a gist linked in the PR description for a summary of changes that was done in the commit, by request of @Lanse505)_

sciwhiz12 committed:
Replace occurences of `World` with `Level`

sciwhiz12 committed:
Replace occurrences of `TileEntity` with `BlockEntity`

sciwhiz12 committed:
Replace occurrences of `Te` with `BlockEntity`

sciwhiz12 committed:
Remove `In` suffix

Contains 13 validation errors, will be fixed in succeeding commits.

sciwhiz12 committed:
Add `Value` suffix to otherwise-invalid parameter names due to keywords

Fixes most of the validation issues, leaving only one error remaining.

sciwhiz12 committed:
Fix parameter name to `pInput`

This was originally `pIn`, but a previous regex-replace commit removed
the `In` suffix, leaving `p` which fails validation. To preserve the
meaning, it's now `pInput`.

Fixes the last validation error.

sciwhiz12 committed:
Replace occurrences of `Placement` to `Decorator`

ChampionAsh5357 commented on data/classes/aet.json:
```diff
@@ -157,7 +157,7 @@
         {
           "index": 1,
-          "name": "pObectIn"
+          "name": "pObect"
```
Object

ChampionAsh5357 commented on data/classes/apg.json:
```diff
@@ -14,7 +14,7 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/apk.json:
```diff
@@ -192,7 +192,7 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/apk.json:
```diff
@@ -350,7 +350,7 @@
         {
           "index": 0,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/apl.json:
```diff
@@ -27,7 +27,7 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/apm.json:
```diff
@@ -20,7 +20,7 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/apo.json:
```diff
@@ -11,11 +11,11 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/apo.json:
```diff
@@ -29,11 +29,11 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/apq.json:
```diff
@@ -11,11 +11,11 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/aps.json:
```diff
@@ -104,7 +104,7 @@
         {
           "index": 3,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/aps.json:
```diff
@@ -122,7 +122,7 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/aps.json:
```diff
@@ -136,11 +136,11 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/aps.json:
```diff
@@ -183,11 +183,11 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/apw$1.json:
```diff
@@ -28,7 +28,7 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/aqa.json:
```diff
@@ -847,11 +847,11 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/aqm.json:
```diff
@@ -739,11 +739,11 @@
         {
           "index": 1,
-          "name": "pLivingentityIn"
+          "name": "pLivingentity"
```
LivingEntity

ChampionAsh5357 commented on data/classes/aqm.json:
```diff
@@ -1551,7 +1551,7 @@
         {
           "index": 1,
-          "name": "pPotioneffectIn"
+          "name": "pPotioneffect"
```
PotionEffect, though it would be either Potion or Effect/EffectInstance or whatever the mojmap equivalent is.

ChampionAsh5357 commented on data/classes/aqm.json:
```diff
@@ -1930,7 +1930,7 @@
         {
           "index": 1,
-          "name": "pItemstackIn"
+          "name": "pItemstack"
```
ItemStack

ChampionAsh5357 commented on data/classes/aqn.json:
```diff
@@ -859,7 +859,7 @@
         {
           "index": 1,
-          "name": "pLeashHolderIDIn"
+          "name": "pLeashHolderID"
```
LeashHolderId

ChampionAsh5357 commented on data/classes/aqn.json:
```diff
@@ -923,7 +923,7 @@
         {
           "index": 1,
-          "name": "pItemstackIn"
+          "name": "pItemstack"
```
ItemStack

ChampionAsh5357 commented on data/classes/aqn.json:
```diff
@@ -1052,7 +1052,7 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/aqs.json:
```diff
@@ -122,7 +122,7 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ayj.json:
```diff
@@ -220,11 +220,11 @@
         {
           "index": 1,
-          "name": "pPathentityIn"
+          "name": "pPathentity"
```
PathEntity, though I'm not sure if `Entity` is needed if it's just referring to the entity's path. Could also be EntityPath or Path.

ChampionAsh5357 commented on data/classes/azj.json:
```diff
@@ -14,7 +14,7 @@
         {
           "index": 0,
-          "name": "pEntitycreatureIn"
+          "name": "pEntitycreature"
```
CreatureEntity

ChampionAsh5357 commented on data/classes/azj.json:
```diff
@@ -99,7 +99,7 @@
         {
           "index": 0,
-          "name": "pEntitycreatureIn"
+          "name": "pEntitycreature"
```
CreatureEntity

ChampionAsh5357 commented on data/classes/azj.json:
```diff
@@ -124,7 +124,7 @@
         {
           "index": 0,
-          "name": "pEntitycreatureIn"
+          "name": "pEntitycreature"
```
CreatureEntity

ChampionAsh5357 commented on data/classes/baf.json:
```diff
@@ -254,7 +254,7 @@
         {
           "index": 1,
-          "name": "pItemstackIn"
+          "name": "pItemstack"
```
ItemStack

ChampionAsh5357 commented on data/classes/bah.json:
```diff
@@ -414,7 +414,7 @@
         {
           "index": 1,
-          "name": "pItemstackIn"
+          "name": "pItemstack"
```
ItemStack

ChampionAsh5357 commented on data/classes/bah.json:
```diff
@@ -506,7 +506,7 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/bal.json:
```diff
@@ -328,7 +328,7 @@
         {
           "index": 1,
-          "name": "pItemstackIn"
+          "name": "pItemstack"
```
ItemStack

ChampionAsh5357 commented on data/classes/bcl.json:
```diff
@@ -346,7 +346,7 @@
         {
           "index": 1,
-          "name": "pPotioneffectIn"
+          "name": "pPotioneffect"
```
PotionEffect, once again either Potion or Effect/EffectInstance

ChampionAsh5357 commented on data/classes/bcn.json:
```diff
@@ -501,7 +501,7 @@
         {
           "index": 1,
-          "name": "pItemstackIn"
+          "name": "pItemstack"
```
ItemStack

ChampionAsh5357 commented on data/classes/bco.json:
```diff
@@ -148,7 +148,7 @@
         {
           "index": 1,
-          "name": "pFacingDirectionIn"
+          "name": "pFacingDirection"
```
Direction? That enum property is used to specify the interaction face or the direction the entity is facing. Though if its a Vector, then this would be correct.

ChampionAsh5357 commented on data/classes/bcq.json:
```diff
@@ -90,7 +90,7 @@
         {
           "index": 1,
-          "name": "pFacingDirectionIn"
+          "name": "pFacingDirection"
```
Direction if enum property, if Vector then should be fine.

ChampionAsh5357 commented on data/classes/bdg.json:
```diff
@@ -277,7 +277,7 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/bdj.json:
```diff
@@ -35,7 +35,7 @@
         {
           "index": 1,
-          "name": "pWololoTargetIn"
+          "name": "pWololoTarget"
```
EvokerSheepTarget or something like that assuming this is referring to the evoker sound made during the sheep color conversion spell.

ChampionAsh5357 commented on data/classes/beb.json:
```diff
@@ -120,7 +120,7 @@
         {
           "index": 1,
-          "name": "pPotioneffectIn"
+          "name": "pPotioneffect"
```
PotionEffect, should be either Potion or Effect/EffectInstance

ChampionAsh5357 commented on data/classes/beh.json:
```diff
@@ -126,7 +126,7 @@
         {
           "index": 1,
-          "name": "pPotioneffectIn"
+          "name": "pPotioneffect"
```
PotionEffect, should be Potion or Effect/EffectInstance

ChampionAsh5357 commented on data/classes/bel.json:
```diff
@@ -236,7 +236,7 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/bqf.json:
```diff
@@ -54,7 +54,7 @@
         {
           "index": 0,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/bsy.json:
```diff
@@ -104,27 +104,27 @@
         {
           "index": 6,
-          "name": "pRandIn"
+          "name": "pRand"
```
Random

ChampionAsh5357 commented on data/classes/bal$a.json:
```diff
@@ -87,7 +87,7 @@
         {
           "index": 0,
-          "name": "pRandIn"
+          "name": "pRand"
```
Random

ChampionAsh5357 commented on data/classes/cee.json:
```diff
@@ -91,7 +91,7 @@
         {
           "index": 2,
-          "name": "pFacingIn"
+          "name": "pFacing"
```
Direction

ChampionAsh5357 commented on data/classes/cgh.json:
```diff
@@ -241,7 +241,7 @@
         {
           "index": 1,
-          "name": "pEntitytypeIn"
+          "name": "pEntitytype"
```
EntityType

ChampionAsh5357 commented on data/classes/cog.json:
```diff
@@ -33,7 +33,7 @@
         {
           "index": 1,
-          "name": "pPlacementBehaviour"
+          "name": "pDecoratorBehaviour"
```
DecoratorBehavior

ChampionAsh5357 commented on data/classes/coi.json:
```diff
@@ -59,7 +59,7 @@
         {
           "index": 1,
-          "name": "pPlacementBehaviour"
+          "name": "pDecoratorBehaviour"
```
DecoratorBehavior

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -101,7 +101,7 @@
         {
           "index": 5,
-          "name": "pBoundingboxIn"
+          "name": "pBoundingbox"
```
BoundingBox

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -114,11 +114,11 @@
         {
           "index": 2,
-          "name": "pBoundingboxIn"
+          "name": "pBoundingbox"
```
BoundingBox

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -166,7 +166,7 @@
         {
           "index": 5,
-          "name": "pBoundingboxIn"
+          "name": "pBoundingbox"
```
BoundingBox

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -210,11 +210,11 @@
         {
           "index": 2,
-          "name": "pBlockstateIn"
+          "name": "pBlockstate"
```
BlockState

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -230,7 +230,7 @@
         {
           "index": 6,
-          "name": "pBoundingboxIn"
+          "name": "pBoundingbox"
```
BoundingBox

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -243,11 +243,11 @@
         {
           "index": 2,
-          "name": "pBoundingboxIn"
+          "name": "pBoundingbox"
```
BoundingBox

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -293,11 +293,11 @@
         {
           "index": 2,
-          "name": "pBoundingboxIn"
+          "name": "pBoundingbox"
```
BoundingBox

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -325,7 +325,7 @@
         {
           "index": 9,
-          "name": "pBlockstateIn"
+          "name": "pBlockstate"
```
BlockState

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -342,11 +342,11 @@
         {
           "index": 2,
-          "name": "pBoundingboxIn"
+          "name": "pBoundingbox"
```
BoundingBox

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -462,11 +462,11 @@
         {
           "index": 2,
-          "name": "pBoundingboxIn"
+          "name": "pBoundingbox"
```
BoundingBox

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -490,7 +490,7 @@
         {
           "index": 8,
-          "name": "pBlockstateIn"
+          "name": "pBlockstate"
```
BlockState

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -613,11 +613,11 @@
         {
           "index": 1,
-          "name": "pBoundingboxIn"
+          "name": "pBoundingbox"
```
BoundingBox

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -665,11 +665,11 @@
         {
           "index": 2,
-          "name": "pBlockstateIn"
+          "name": "pBlockstate"
```
BlockState

ChampionAsh5357 commented on data/classes/cru.json:
```diff
@@ -685,7 +685,7 @@
         {
           "index": 6,
-          "name": "pBoundingboxIn"
+          "name": "pBoundingbox"
```
BoundingBox

ChampionAsh5357 commented on data/classes/csw.json:
```diff
@@ -65,11 +65,11 @@
         {
           "index": 2,
-          "name": "pExtIn"
+          "name": "pExt"
```
Extension

ChampionAsh5357 commented on data/classes/csw.json:
```diff
@@ -83,11 +83,11 @@
         {
           "index": 2,
-          "name": "pExtIn"
+          "name": "pExt"
```
Extension

ChampionAsh5357 commented on data/classes/cxa.json:
```diff
@@ -80,7 +80,7 @@
         {
           "index": 1,
-          "name": "pBlockaccessIn"
+          "name": "pBlockaccess"
```
BlockAccess, is that what this class is still called?

ChampionAsh5357 commented on data/classes/cxa.json:
```diff
@@ -96,7 +96,7 @@
         {
           "index": 5,
-          "name": "pEntitylivingIn"
+          "name": "pEntityliving"
```
LivingEntity

ChampionAsh5357 commented on data/classes/cxb.json:
```diff
@@ -105,7 +105,7 @@
         {
           "index": 1,
-          "name": "pPathpointIn"
+          "name": "pPathpoint"
```
PathPoint

ChampionAsh5357 commented on data/classes/cxb.json:
```diff
@@ -148,7 +148,7 @@
         {
           "index": 1,
-          "name": "pPathpointIn"
+          "name": "pPathpoint"
```
PathPoint

ChampionAsh5357 commented on data/classes/cxc.json:
```diff
@@ -114,7 +114,7 @@
         {
           "index": 1,
-          "name": "pBlockaccessIn"
+          "name": "pBlockaccess"
```
BlockAccess, is that what this class is still called?

ChampionAsh5357 commented on data/classes/cxc.json:
```diff
@@ -130,7 +130,7 @@
         {
           "index": 5,
-          "name": "pEntitylivingIn"
+          "name": "pEntityliving"
```
LivingEntity

ChampionAsh5357 commented on data/classes/cxd.json:
```diff
@@ -112,7 +112,7 @@
         {
           "index": 1,
-          "name": "pPathentityIn"
+          "name": "pPathentity"
```
PathEntity, though it should probably be EntityPath or just Path.

ChampionAsh5357 commented on data/classes/cxg.json:
```diff
@@ -70,7 +70,7 @@
         {
           "index": 1,
-          "name": "pBlockaccessIn"
+          "name": "pBlockaccess"
```
BlockAccess, is that what this class is still called?

ChampionAsh5357 commented on data/classes/cxg.json:
```diff
@@ -86,7 +86,7 @@
         {
           "index": 5,
-          "name": "pEntitylivingIn"
+          "name": "pEntityliving"
```
LivingEntity

ChampionAsh5357 commented on data/classes/cxj.json:
```diff
@@ -102,7 +102,7 @@
         {
           "index": 1,
-          "name": "pEntitylivingIn"
+          "name": "pEntityliving"
```
LivingEntity

ChampionAsh5357 commented on data/classes/cxj.json:
```diff
@@ -201,7 +201,7 @@
         {
           "index": 1,
-          "name": "pBlockaccessIn"
+          "name": "pBlockaccess"
```
BlockAccess, is that what this class is still called?

ChampionAsh5357 commented on data/classes/cxj.json:
```diff
@@ -217,7 +217,7 @@
         {
           "index": 5,
-          "name": "pEntitylivingIn"
+          "name": "pEntityliving"
```
LivingEntity

ChampionAsh5357 commented on data/classes/cyc.json:
```diff
@@ -71,7 +71,7 @@
         {
           "index": 2,
-          "name": "pWorldVersion"
+          "name": "pLevelVersion"
```
I think this is wrong because WorldVersion iirc refers to the actual version data used by `DataFixer` to fix data between mc versions.

ChampionAsh5357 commented on data/classes/db.json:
```diff
@@ -181,7 +181,7 @@
         {
           "index": 1,
-          "name": "pPitchYawIn"
+          "name": "pPitchYaw"
```
Aren't these two different things? Not sure why they would be on the same parameter.

ChampionAsh5357 commented on data/classes/dcj.json:
```diff
@@ -44,11 +44,11 @@
         {
           "index": 1,
-          "name": "pFaceIn"
+          "name": "pFace"
```
Direction most likely.

ChampionAsh5357 commented on data/classes/dkx.json:
```diff
@@ -43,7 +43,7 @@
         {
           "index": 1,
-          "name": "pMapdataIn"
+          "name": "pMapdata"
```
MapData

ChampionAsh5357 commented on data/classes/dkx.json:
```diff
@@ -110,7 +110,7 @@
         {
           "index": 1,
-          "name": "pMapdataIn"
+          "name": "pMapdata"
```
MapData

ChampionAsh5357 commented on data/classes/dlq.json:
```diff
@@ -250,7 +250,7 @@
         {
           "index": 1,
-          "name": "pRssponderIn"
+          "name": "pRssponder"
```
Responder

ChampionAsh5357 commented on data/classes/dyg.json:
```diff
@@ -179,15 +179,15 @@
         {
           "index": 1,
-          "name": "pParticleRedIn"
+          "name": "pParticleRed"
```
Red, sets the color in the `Particle` class so doesn't need prefix.

ChampionAsh5357 commented on data/classes/dyg.json:
```diff
@@ -179,15 +179,15 @@
         {
           "index": 2,
-          "name": "pParticleGreenIn"
+          "name": "pParticleGreen"
```
Green, sets the color in the `Particle` class so doesn't need prefix.

ChampionAsh5357 commented on data/classes/dyg.json:
```diff
@@ -179,15 +179,15 @@
         {
           "index": 3,
-          "name": "pParticleBlueIn"
+          "name": "pParticleBlue"
```
Blue, sets the color in the `Particle` class so doesn't need prefix.

ChampionAsh5357 commented on data/classes/dzz.json:
```diff
@@ -265,11 +265,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eae$a.json:
```diff
@@ -41,7 +41,7 @@
         {
           "index": 2,
-          "name": "pFacingIn"
+          "name": "pFacing"
```
Direction

ChampionAsh5357 commented on data/classes/eae$a.json:
```diff
@@ -55,7 +55,7 @@
         {
           "index": 1,
-          "name": "pFacingIn"
+          "name": "pFacing"
```
Direction

ChampionAsh5357 commented on data/classes/eaz$c.json:
```diff
@@ -69,7 +69,7 @@
         {
           "index": 0,
-          "name": "pFacingIn"
+          "name": "pFacing"
```
Direction

ChampionAsh5357 commented on data/classes/eaz.json:
```diff
@@ -29,31 +29,31 @@
         {
           "index": 1,
-          "name": "pBlockAccessIn"
+          "name": "pBlockAccess"
```
Pretty sure this is not what the class is called anymore.

ChampionAsh5357 commented on data/classes/eaz.json:
```diff
@@ -75,19 +75,19 @@
         {
           "index": 1,
-          "name": "pBlockAccessIn"
+          "name": "pBlockAccess"
```
Pretty sure this is not what the class is called anymore.

ChampionAsh5357 commented on data/classes/eaz.json:
```diff
@@ -121,15 +121,15 @@
         {
           "index": 1,
-          "name": "pBlockAccessIn"
+          "name": "pBlockAccess"
```
Pretty sure this is not what the class is called anymore.

ChampionAsh5357 commented on data/classes/ebh.json:
```diff
@@ -59,19 +59,19 @@
         {
           "index": 0,
-          "name": "pBlockFaceUVIn"
+          "name": "pBlockFaceUV"
```
Need to determine consistency for UV vs Uv. Seen param names as Uvs.

ChampionAsh5357 commented on data/classes/ebh.json:
```diff
@@ -239,7 +239,7 @@
         {
           "index": 4,
-          "name": "pBlockFaceUVIn"
+          "name": "pBlockFaceUV"
```
Need to determine consistency UV vs Uv.

ChampionAsh5357 commented on data/classes/ecu$c$b.json:
```diff
@@ -65,15 +65,15 @@
         {
           "index": 2,
-          "name": "pTileEntitiesIn"
+          "name": "pTileEntities"
```
BlockEntities

sciwhiz12 replied:
Manually fixed this in-scope change.

ChampionAsh5357 commented on data/classes/edv.json:
```diff
@@ -32,11 +32,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eea.json:
```diff
@@ -79,11 +79,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eee.json:
```diff
@@ -69,11 +69,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eef.json:
```diff
@@ -58,11 +58,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eek.json:
```diff
@@ -72,11 +72,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eeo.json:
```diff
@@ -58,11 +58,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ees.json:
```diff
@@ -30,7 +30,7 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eew$1.json:
```diff
@@ -17,19 +17,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eew$1.json:
```diff
@@ -63,19 +63,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eew$1.json:
```diff
@@ -109,19 +109,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efc.json:
```diff
@@ -62,11 +62,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efd.json:
```diff
@@ -62,11 +62,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efe.json:
```diff
@@ -87,7 +87,7 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efi.json:
```diff
@@ -67,11 +67,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efj.json:
```diff
@@ -26,11 +26,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efk$1.json:
```diff
@@ -17,19 +17,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efk$1.json:
```diff
@@ -63,19 +63,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efk$1.json:
```diff
@@ -109,19 +109,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efo.json:
```diff
@@ -167,11 +167,11 @@
         {
           "index": 3,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efr.json:
```diff
@@ -104,11 +104,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efr.json:
```diff
@@ -209,7 +209,7 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efu.json:
```diff
@@ -70,11 +70,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/efw.json:
```diff
@@ -110,19 +110,19 @@
         {
           "index": 1,
-          "name": "pEntityLivingIn"
+          "name": "pEntityLiving"
```
LivingEntity

ChampionAsh5357 commented on data/classes/egc.json:
```diff
@@ -79,11 +79,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/egg.json:
```diff
@@ -58,11 +58,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/egp.json:
```diff
@@ -30,7 +30,7 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/egr.json:
```diff
@@ -160,11 +160,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/egu.json:
```diff
@@ -43,7 +43,7 @@
         {
           "index": 1,
-          "name": "pEntityLivingBaseIn"
+          "name": "pEntityLivingBase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/egx.json:
```diff
@@ -66,11 +66,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehg.json:
```diff
@@ -83,11 +83,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehh.json:
```diff
@@ -58,11 +58,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehi$1.json:
```diff
@@ -17,19 +17,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehi$1.json:
```diff
@@ -63,19 +63,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehi$1.json:
```diff
@@ -109,19 +109,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehj.json:
```diff
@@ -58,11 +58,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehk.json:
```diff
@@ -160,11 +160,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehl.json:
```diff
@@ -74,11 +74,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehm.json:
```diff
@@ -67,11 +67,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehu.json:
```diff
@@ -47,19 +47,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehv.json:
```diff
@@ -47,19 +47,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehw.json:
```diff
@@ -61,19 +61,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehy.json:
```diff
@@ -47,19 +47,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ehz.json:
```diff
@@ -65,19 +65,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eia.json:
```diff
@@ -47,19 +47,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eib.json:
```diff
@@ -47,19 +47,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eic.json:
```diff
@@ -61,19 +61,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eif.json:
```diff
@@ -19,19 +19,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eig.json:
```diff
@@ -15,19 +15,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eih.json:
```diff
@@ -47,19 +47,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eii.json:
```diff
@@ -53,19 +53,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eij.json:
```diff
@@ -57,19 +57,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eik.json:
```diff
@@ -73,19 +73,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eil.json:
```diff
@@ -57,19 +57,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eim.json:
```diff
@@ -47,19 +47,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ein.json:
```diff
@@ -51,19 +51,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eio.json:
```diff
@@ -65,19 +65,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eiq.json:
```diff
@@ -47,19 +47,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eir.json:
```diff
@@ -57,19 +57,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eir.json:
```diff
@@ -103,19 +103,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eit.json:
```diff
@@ -27,19 +27,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eiu.json:
```diff
@@ -21,19 +21,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eiv.json:
```diff
@@ -61,19 +61,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eiw.json:
```diff
@@ -47,19 +47,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eix.json:
```diff
@@ -53,19 +53,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eiy.json:
```diff
@@ -47,19 +47,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eja.json:
```diff
@@ -61,19 +61,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ejb.json:
```diff
@@ -61,19 +61,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ejc.json:
```diff
@@ -55,19 +55,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ejd.json:
```diff
@@ -57,19 +57,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/eje.json:
```diff
@@ -83,19 +83,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ejf.json:
```diff
@@ -47,19 +47,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ejh.json:
```diff
@@ -57,19 +57,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/ejk.json:
```diff
@@ -254,11 +254,11 @@
         {
           "index": 1,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 commented on data/classes/enq.json:
```diff
@@ -47,7 +47,7 @@
         {
           "index": 1,
-          "name": "pSoundIDIn"
+          "name": "pSoundID"
```
SoundId

ChampionAsh5357 commented on data/classes/gv.json:
```diff
@@ -74,7 +74,7 @@
         {
           "index": 2,
-          "name": "pFacingIn"
+          "name": "pFacing"
```
Direction

ChampionAsh5357 commented on data/classes/mj.json:
```diff
@@ -120,7 +120,7 @@
         {
           "index": 1,
-          "name": "pIIn"
+          "name": "pI"
```
Index

ChampionAsh5357 commented on data/classes/mj.json:
```diff
@@ -167,7 +167,7 @@
         {
           "index": 1,
-          "name": "pIIn"
+          "name": "pI"
```
Index

ChampionAsh5357 commented on data/classes/mj.json:
```diff
@@ -185,7 +185,7 @@
         {
           "index": 1,
-          "name": "pIIn"
+          "name": "pI"
```
Index

ChampionAsh5357 commented on data/classes/bfg.json:
```diff
@@ -100,7 +100,7 @@
         {
           "index": 1,
-          "name": "pXpIn"
+          "name": "pXp"
```
Experience, or other Experience parameters should be adjusted to Xp.

ChampionAsh5357 commented on data/classes/bfe.json:
```diff
@@ -303,7 +303,7 @@
         {
           "index": 1,
-          "name": "pXpIn"
+          "name": "pXp"
```
Experience, or other Experience parameters should be adjusted to Xp.

ChampionAsh5357 commented on data/classes/bfj.json:
```diff
@@ -713,7 +713,7 @@
         {
           "index": 1,
-          "name": "pXpIn"
+          "name": "pXp"
```
Experience, or other Experience parameters should be adjusted to Xp.

ChampionAsh5357 commented on data/classes/bqu.json:
```diff
@@ -100,7 +100,7 @@
         {
           "index": 1,
-          "name": "pXpIn"
+          "name": "pXp"
```
Experience, or other Experience parameters should be adjusted to Xp.

ChampionAsh5357 commented on data/classes/ebe.json:
```diff
@@ -31,7 +31,7 @@
         {
           "index": 1,
-          "name": "pUvsIn"
+          "name": "pUvs"
```
Needs consistency UV vs Uv.

ChampionAsh5357 commented on data/classes/ebf.json:
```diff
@@ -271,7 +271,7 @@
         {
           "index": 0,
-          "name": "pStrIn"
+          "name": "pStr"
```
String, or standardized to Str. In either case it needs to be more descriptive.

ChampionAsh5357 commented on data/classes/py.json:
```diff
@@ -60,7 +60,7 @@
         {
           "index": 1,
-          "name": "pMapdataIn"
+          "name": "pMapdata"
```
MapData

ChampionAsh5357 commented on data/classes/eid.json:
```diff
@@ -61,19 +61,19 @@
         {
           "index": 4,
-          "name": "pEntitylivingbaseIn"
+          "name": "pEntitylivingbase"
```
LivingEntity

ChampionAsh5357 reviewed (approved):
This is a full audit of the changed parameter names as requested by sci.
Total review time: 4:27:14.27

**There are no issues with the regexes mentioned in the above PR.** Most of these are just a bunch of general comments about the MCP names in terms of formatting, capitalization, and current class name. I didn't check any names that weren't changed by this PR since otherwise I would be here for longer than 5 hours.

In-scope cases that weren't covered by the regex:
ecu$c$b: TileEntities -> BlockEntities

ChampionAsh5357 reviewed (approved):
This is a full audit of the changed parameter names as requested by sci.
Total review time: 4:27:14.27

**There are no issues with the regexes mentioned in the above PR.** Most of these are just a bunch of general comments about the MCP names in terms of formatting, capitalization, and current class name. I didn't check any names that weren't changed by this PR since otherwise I would be here for longer than 5 hours.

In-scope cases that weren't covered by the regex:
ecu$c$b: TileEntities -> BlockEntities

sciwhiz12 committed:
Fix unmatched `pTileEntities` to `pBlockEntities` manually

sciwhiz12 committed:
Replace occurences of `EntityLivingBase` to `LivingEntity`

ChampionAsh5357 reviewed (approved):
Covers all the base regexes provided now.

sciwhiz12 commented:
Under the presumption that our other reviewers won't review this PR due to the massive amount of changes (which led to @ChampionAsh5357 taking _4+ hours to review_), and that because this PR consists mostly of regex-driven changes (bar a few manual fixes), I am going to mark this as merge-ready and will merge this in a few minutes.

After this, we will then move the 1.16.5 data to 1.17 (using MCPConfig's SRG as the intermediary), and focus future work on 1.17.
